src/main/java/org/broadinstitute/hellbender/utils/pairhmm/SampleSitesForDRAGstrModel.java

```diff
@@ -0,0 +1,437 @@
+                return masks[repeats];
+            }
+        }
+
+        public boolean decimate(final int seqNumber, final int bestPeriod, final long bestPeriodRepeats) {
+            if (counts.length <= bestPeriod) {
+                return false;
+            } else {
+                final long[] periodCounts = counts[bestPeriod];
+                if (periodCounts.length == 0) {
+                    return false;
+                } else {
+                    final int effectiveRepeatCount
+                            = (int) (bestPeriodRepeats < periodCounts.length ? bestPeriodRepeats : periodCounts.length - 1);
+                    final long count = periodCounts[effectiveRepeatCount]++;
+                    final long left = count + seqNumber;
+                    final long right = decimationMask[bestPeriod][effectiveRepeatCount];
+                    return ((int) left & (int) right) != 0 || ((left >> 32) & (right >> 32)) != 0;
+                }
+            }
+        }
+    }
+
+    @Argument(fullName="decimation", doc="decimation per period and repeat. It can be \"DEFAULT\" to use the default values (default), " +
+            " \"NONE\" to deactivate decimation (potentially resulting in a very large output file) or indicate the path to a file" +
+            " that contains the decimation matrix.", optional = true)
+    private DecimationTable decimationTable = DecimationTable.DEFAULT;
+
+    @Argument(doc = "name of the zip file where the sites sampled will be stored",
+              fullName= StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+              shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME)
+    private String outputPath = null;
+
+    @Argument(fullName="max-period", doc="maximum STR period sampled", optional = true, minValue = 1, maxValue = 10)
+    private int maxPeriod = 8;
+
+    @Argument(fullName="max-repeats", doc="maximum STR repeat sampled", optional = true, minValue = 1, maxValue = 20)
+    private int maxRepeat = 20;
+
+    @Override
+    public boolean requiresReference() {
+        return true;
+    }
+
+    private File tempDir;
+
+    private long[][] totalCounts;
+
+    private long[][] emittedCounts;
+
+    public void onStartup() {
+        super.onStartup();
+        try {
+            tempDir = File.createTempFile("gatk-sample-dragstr-sites", ".tmp");
+        } catch (final IOException ex) {
+            throw new GATKException("could not create temporary disk space", ex);
+        }
+        if (!tempDir.delete()) {
+            throw new GATKException("could not create temporary disk space: could not delete tempfile");
+        } else if (!tempDir.mkdir()) {
+            throw new GATKException("could not create temporary disk space: could not create tempdir");
+        }
+        totalCounts = new long[maxPeriod + 1][maxRepeat + 1];
+        emittedCounts = new long[maxPeriod + 1][maxRepeat + 1];
+    }
+
+    public void onShutdown() {
+        Utils.deleteFileTree(tempDir);
```

Comment:
Use a system directory delete option here. Also add a check that the tmp dir was never full to begin with at the start.